test/WebResourceTest.js

```diff
@@ -0,0 +1,61 @@
+import * as helpers from './helpers';
+import * as chai from 'chai';
+import * as chaiDateTime from 'chai-datetime';
+import {
+	getStorageAdapter,
+	registerStorageAdapter,
+} from '../out/storage-adapters';
+import {
+	DISK_STORAGE_ADAPTER_NAME,
+	DiskStorageAdapter,
+} from './storage-adapters/disk-storage-adapter';
+
+chai.use(chaiDateTime);
+const { expect } = chai;
+
+registerStorageAdapter(DISK_STORAGE_ADAPTER_NAME, DiskStorageAdapter);
+
+const hex = '5261777221';
+const buf = Buffer.from(hex, 'hex');
+
+const webResourceInput = {
+	filename: 'logo.png',
+	data: buf,
+	contentType: 'image/png',
+	contentDisposition: null,
+	size: buf.byteLength,
+	storage: DISK_STORAGE_ADAPTER_NAME,
+};
```

Comment:
Since this repo is mainly a pinejs component, the optional properties are set by pineJS by taking the properties from `multer` and `filetype`. `content-type` is validated and set up by pinejs, `size` is picked up from multer. `content-disposition` is not used.

I've added a test case, `validate` works with them being `undefined`.